packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1024,36 +1025,52 @@ export class Stack extends Construct implements ITaggable {
-    const exportable = getExportable(this, resolvable);
+    const importValue = this.determineImportValue(exportedValue);
 
-    // Ensure a singleton "Exports" scoping Construct
-    // This mostly exists to trigger LogicalID munging, which would be
-    // disabled if we parented constructs directly under Stack.
-    // Also it nicely prevents likely construct name clashes
-    const exportsScope = getCreateExportsScope(this);
+    if (Array.isArray(importValue)) {
+      throw new Error('Attempted to export a list value from `exportValue()`: use `exportStringListValue()` instead');
+    }
 
-    // Ensure a singleton CfnOutput for this value
-    const resolved = this.resolve(exportable);
-    const id = 'Output' + JSON.stringify(resolved);
-    const exportName = generateExportName(exportsScope, id);
+    return importValue;
+  }
 
-    if (Token.isUnresolved(exportName)) {
-      throw new Error(`unresolved token in generated export name: ${JSON.stringify(this.resolve(exportName))}`);
+  /**
+   * Create a CloudFormation Export for a string list value
+   *
+   * Returns a string list representing the corresponding `Fn.importValue()`
+   * expression for this Export. The export expression is automatically wrapped with an
+   * `Fn::Join` and the import value with an `Fn::Split`, since CloudFormation can only
+   * export strings. You can control the name for the export by passing the `name` option.
+   *
+   * If you don't supply a value for `name`, the value you're exporting must be
+   * a Resource attribute (for example: `bucket.bucketName`) and it will be
+   * given the same name as the automatic cross-stack reference that would be created
+   * if you used the attribute in another Stack.
+   *
+   * One of the uses for this method is to *remove* the relationship between
+   * two Stacks established by automatic cross-stack references. It will
+   * temporarily ensure that the CloudFormation Export still exists while you
+   * remove the reference from the consuming stack. After that, you can remove
+   * the resource and the manual export.
+   *
+   * # See `exportValue` for an example of this process.
+   */
+  public exportStringListValue(exportedValue: any, options: ExportValueOptions = {}): string[] {
+    if (options.name) {
+      new CfnOutput(this, `Export${options.name}`, {
+        value: Fn.join(STRING_LIST_REFERENCE_DELIMITER, exportedValue),
+        exportName: options.name,
+      });
+      return Fn.split(STRING_LIST_REFERENCE_DELIMITER, Fn.importValue(options.name));
     }
 
-    const output = exportsScope.node.tryFindChild(id) as CfnOutput;
-    if (!output) {
-      new CfnOutput(exportsScope, id, { value: Token.asString(exportable), exportName });
+    const importValue = this.determineImportValue(exportedValue);
+
+    if (!Array.isArray(importValue)) {
+      throw new Error('Attempted to export a string value from `exportStringListValue()`: use `exportValue()` instead');
```

Comment:
For some reason this makes me uneasy. Can you do inspections on the value's type and type hints, before we go to export it, instead?